server/v1/reader.go

```diff
@@ -32,18 +33,26 @@ import (
 )
 
 type reader struct {
-	in      *bufio.Reader
-	conn    net.Conn
-	timeout time.Duration
-	buf     []byte
+	conn       net.Conn
+	in         *bufio.Reader
+	tlsState   *tls.ConnectionState
+	remoteAddr string
+	buf        []byte
```

Comment:
If we are confident that it's nearly always the case that the `buf` is not grown beyond 64 bytes long, maybe make it be a slice of another `[64]byte` field?

Reply:
I'm not confident on this so I'm going to leave it as is. But good idea.